bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/StringListType.java

```diff
@@ -38,12 +38,16 @@ public StringListType() {
     }
 
     public StringListType(StringType... rows) {
-        typeDetails = new ArrayList<String>(rows.length);
+        typeDetails = new ArrayList<>(rows.length);
         for (StringType row : rows) {
             typeDetails.add(row.toString());
         }
     }
 
+    public StringListType(List<String> rows) {
+        this.typeDetails = new ArrayList<>(rows);
```

Comment:
Would it be possible to let the convenience constructors delegate to one constructor which is responsible for setting the `typeDetails`? My suggestion is to use your new c`tor and let all the others delegate to it.